### PR TITLE
DAOS-11411 vos: Add new metrics to track checkpointing activity

### PR DESCRIPTION
**Pull request by sherintg**
Added metrics to capture the checkpoint duration, pages copied and wal size purged.

Additionally fixed a bug related to mismatch in meta blob size and the vos file size.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by jolivier23 — DISMISSED** [review]

**Review by NiuYawei — CHANGES_REQUESTED** [review]

File: src/common/dav/dav_iface.c [hunk 1]
```diff
@@ -294,7 +296,15 @@ dav_obj_create(const char *path, int flags, size_t sz, mode_t mode, struct umem_
...
 	}
 
-	hdl = dav_obj_open_internal(fd, DAV_HEAP_INIT, sz, path, store);
+	if (sz < store->stor_size) {
+		ERR("size argument cannot be less then the actual file size\n");
+		errno = EINVAL;
+		close(fd);
+		return NULL;
+	}
+	D_ASSERT(sz >= store->stor_size);
```
[the remark this answers is not in the document]
Comment (sherintg) [reply src/common/dav/dav_iface.c]: Fixed

File: src/common/dav/dav_iface.c [hunk 2]
```diff
@@ -294,7 +296,15 @@ dav_obj_create(const char *path, int flags, size_t sz, mode_t mode, struct umem_
 		}
 	}
 
-	hdl = dav_obj_open_internal(fd, DAV_HEAP_INIT, sz, path, store);
+	if (sz < store->stor_size) {
+		ERR("size argument cannot be less then the actual file size\n");
```
[the remark this answers is not in the document]
Comment (sherintg) [reply src/common/dav/dav_iface.c]: ERR is mapped to D_ERROR. The reason no error was printed was due to stor_size being zero and hence the condition passed. I have modified the condition to catch stor_size == 0 aswell.

File: src/include/daos_srv/bio.h [hunk 3]
```diff
@@ -1029,10 +1029,11 @@ int bio_wal_flush_header(struct bio_meta_context *mc);
  *
  * \param[in]	mc		BIO meta context
  * \param[in]	tx_id		The highest checkpointed transaction ID
+ * \param[out]	purge_size	wal size purged
```
[the remark this answers is not in the document]
Comment (sherintg) [reply src/include/daos_srv/bio.h]: Fixed

File: src/common/tests/umem_test_bmem.c [hunk 4]
```diff
@@ -73,7 +73,7 @@ struct umem_store_ops _store_ops = {
 	.so_wal_submit = _persist_submit,
 };
 
-struct umem_store ustore = { .stor_ops = &_store_ops };
+struct umem_store ustore = { .stor_size = POOL_SIZE, .stor_ops = &_store_ops };
```
Comment (NiuYawei) [inline src/common/tests/umem_test_bmem.c:76]: The correct stor_size is "POOL_SIZE - 4k", won't it cause bmem unit test failure by specifying slightly larger size?

**Review by NiuYawei — APPROVED** [review]

File: src/common/tests/umem_test_bmem.c (hunk 4 above)
Comment (sherintg) [reply src/common/tests/umem_test_bmem.c:76]: The current fix works as long as stor_size <= sz condition is true.

**Review by jolivier23 — APPROVED** [review]

**Comment by sherintg** [comment]
Got confirmation from @NiuYawei that these are know failures - DAOS-12883 and DAOS-12922

**Pull request merged**